lib/doc/main.js

```diff
@@ -68,10 +68,23 @@ function isTopLink(elem) {
   return elem.getAttribute('href') === '#';
 }
 
+function isAnchorLink(elem) {
+  var href = elem.getAttribute('href');
+  return elem.tagName === 'A' && href.length > 1 && href.substr(0, 1) === '#';
```

Comment:
If we remove the `href.length > 1` part, it will also trigger on `href="#"`, whereas before it wouldn't have. That was intentional.

However, the only link I could find with that exact href attribute is the Documentation menu item, and I'd like to see the 'close menu' behaviour there as well, so the change makes sense.

I've updated the PR.